### ISSUE TITLE
[preset-env] target.node=true should use package.engines.node

### ISSUE THREAD

**Issue by julien-f** [babel/babel#7277]
### Is this a bug report or feature request?

Feature request

### Input Code

```js
var your => (code) => here;
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```js
{
  presets: ['env', { targets: { node: true } }]
}
```

`package.json`:

```json
{
  "engines": {
    "node": ">=6"
  }
}
```

### Expected Behavior

It should target the Node version specified in `package.engines.node`.

### Current Behavior

It targets the current Node version.

### Possible Solution

`current` should still use the current version, but `true` could use `package.engines.node`.

### Context

I currently have to maintain in sync the targeted Node version both in `package.json` and in the Babel config.

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  
| Babylon          | 
| node             | 
| npm              | 
| Operating System |

**Comment by Andarist** [comment]
Wouldnt `node: "engines"` be more explicit and readable? Although it's also worth discussing if `engines` could support also some format for `browsers` - and if yes then this should be a top-level option and not only for `node`.

**Comment by julien-f** [comment]
`engines` is already a standard field in `package.json`, see [the documentation](https://docs.npmjs.com/files/package.json#engines).

Concerning the browsers, `preset-env` uses `browserslist` which does expose a pseudo-standard `browserslist` field ([doc](https://github.com/ai/browserslist#browserslist)).

**Comment by Andarist** [comment]
I know its a standard field, but as it is an object with "node" key and it could easily support browsers too - but i guess it could have been more useful for other things (i.e. when consuming packages with specified `engines.browsers`) and is not tightly related to your request

**Comment by julien-f** [comment]
Yes, it could support browsers as well, but this probably an issue that should be created in the `browserslist` repo :)

**Comment by existentialism** [comment]
For history/reference: https://github.com/babel/babel-preset-env/pull/114

**Comment by jaydenseric** [comment]
It should just read from `engines` by default, like it reads from `browserslist` by default. Zero configuration FTW!

That you have to set this up manually is my single biggest frustration with present-env. [In .babelrc.js I require package.json and pull in `engines.node`](https://github.com/jaydenseric/apollo-upload-client/blob/v7.0.0-alpha.4/.babelrc.js#L1).

It is shocking how many published packages don't follow best practice to define `engines.node`; making it the default way to configure preset-env could really help this situation.

**Comment by julien-f** [comment]
Agreed, but this should be overridable (maybe with `node: false`).

**Comment by grantlouisherman** [comment]
Is this ticket still open?

**Comment by xtuc** [comment]
@grantlouisherman yes

**Comment by grantlouisherman** [comment]
@xtuc I would like to take it if possible.

**Comment by Leko** [comment]
@existentialism @xtuc
Hi there. I try to tackle this issue.
I have two questions.

1. What the best way to lookup `engines` field in package.json?
I think the better to use `findPackageData` in `@babel/core` but it is private function.
May I use `fs` to lookup engines filed?

2. What behavior when engines field not defined
Should we report error immediately or ignore silently or fallback to `semverify`?

**Comment by DZuz14** [comment]
Is the code responsible for this located here? 
https://github.com/babel/babel/blob/master/packages/babel-preset-env/src/targets-parser.js#L145

**Comment by existentialism** [comment]
You can reference a previous PR for this over at the old repo: https://github.com/babel/babel-preset-env/pull/114.

Keep in mind, there were some unresolved questions in that thread that we might want to nail down :)

**Comment by DZuz14** [comment]
Excuse my naivety on this issue(and the Babel project in general), but why cant the node field in targets just have a simple true or false value, and require that people set their node version in package.engines.node in their package.json?

In the package.json, you'd have to explicitly specify "current", ">=6"  etc.

**Comment by dhruvdutt** [comment]
Is this still available to work on?

**Comment by julien-f** [comment]
I believe so.

@existentialism do you confirm?

**Comment by niftylettuce** [comment]
What would be really nice is to be able to specify the `node` targets in `package.json`, e.g. `nodeTargets`, and also make use of `browserlist` key in `package.json` as well.  Right now if you specify a `targets` option in `.babelrc` and also have `.browserlistrc` file, the `.browserlistrc` file will not be read.  In some cases such as using `eslint` with plugins like `eslint-plugin-compat` (to check for browser compatibility using caniuse database) - you always need to specify the same `browserlist` string...  It's quite ugly.

**Comment by pree-T** [comment]
May I work on it? Can someone help me as I am new to open source contribution?

**Comment by LeonxLJX** [comment]
I'd like to take this one.

I've read through the report. Before changing anything I'd like to reproduce it against current `main` first — my reasoning is that a fix anchored to the actual observed behavior is much easier to review than one inferred from the report alone, especially if the surrounding code has moved since this was filed.

Plan: reproduce and confirm the root cause, then make the smallest change that addresses it (plus a regression test if the repo has coverage for this area). If the root cause turns out to live somewhere other than where the report points, I'll report back here before forcing a change.

Anything already known about this that I should read first?

(claiming via @LeonxLJX)
